test/src/androidTest/kotlin/io/realm/shared/RealmListTests.kt

```diff
@@ -806,6 +827,91 @@ internal abstract class ManagedListTester<T>(
         }
     }
 
+    override fun observeEmitListOnListCollect() {
+        val dataSet = typeSafetyManager.getInitialDataSet()
+
+        errorCatcher {
+            realm.writeBlocking {
+                val list = typeSafetyManager.createContainerAndGetList(this)
+                list.addAll(copyToRealmIfNeeded(dataSet))
+            }
+
+            runBlocking {
+                val channel = Channel<RealmList<*>>(capacity = 1)
+                val observer = async {
+                    val container = realm.objects<RealmListContainer>()
+                        .first()
+
+                    typeSafetyManager.getList(container)
+                        .observe()
+                        .onEach { flowList ->
```

Comment:
Why sending this as a side effect instead of inside the `collect` method. IMO that would be more readable:

```
typeSafetyManager.getList(container)
                        .observe()
                        .collect { flowList ->
                             channel.send(flowList)
                         }
```